tests/test_samples.py

```diff
@@ -424,6 +434,21 @@ def test_masking():
         pc[mask].plot_2d(axes=axes, types=dict(lower=ptype, diagonal='hist'))
 
 
+def test_merging():
+    numpy.random.seed(3)
+    samples_1 = NestedSamples(root='./tests/example_data/pc')
+    samples_2 = NestedSamples(root='./tests/example_data/pc_250')
+    samples = merge_nested_samples([samples_1, samples_2])
+    nlive_1 = samples_1.nlive.mode()[0]
+    nlive_2 = samples_2.nlive.mode()[0]
+    nlive = samples.nlive.mode()[0]
+    assert nlive == nlive_1 + nlive_2
```

Comment:
I know there is the test `test_utils.test_compute_nlive`, however, there the test data is generated by `numpy.random.rand` and that seems somewhat simplified compared to PolyChord output. 

This might be just me not as familiar with the code, but maybe it would be worth to **add a test that checks whether `samples_1.nlive.mode()[0]` is indeed what we expect from the example data?**

Reply:
I'm also unsure at a glance, why `.mode()` is the right thing to use to get `nlive`. Is that because throughout a long PC run the number of live points will almost always be `nlive`?

Reply:
Does [the latest commit](https://github.com/williamjameshandley/anesthetic/pull/61/commits/70842b1538bfffaf3a6ef02e3bf8d2e473b83645) give you the functionality you were expecting?

Reply:
> I'm also unsure at a glance, why .mode() is the right thing to use to get nlive. Is that because throughout a long PC run the number of live points will almost always be nlive?

For these runs, nlive starts at e.g. 125, remains constant until the very end, where then it drops linearly due to the impact of the final set of live points:

![Figure_1](https://user-images.githubusercontent.com/3708486/69550082-542b7f00-0f92-11ea-8136-9aadce9c99c0.png)

The mode is just a quick-and-easy way to get the nlive of the main part of the run

Reply:
Nice plot, I like how you can see the different drop offs in the combined sample. 

Ok, this makes sense then and I do like the hard coded numbers in there. Though not great in the code itself they do have a place in unit testing.

Reply:
Pretty easy to do as well:
```python
    samples_1 = NestedSamples(root='./tests/example_data/pc')
    samples_2 = NestedSamples(root='./tests/example_data/pc_250')
    samples = merge_nested_samples([samples_1, samples_2])

    samples_1.nlive.plot(label='samples_1')
    samples_2.nlive.plot(label='samples_2')
    ax = samples.nlive.plot(label='samples')
    ax.set_ylabel('nlive')
    ax.legend()

```

Reply:
Here's an alternative view on it, plotting against compression rather than iteration

```python
    samples_1 = NestedSamples(root='./tests/example_data/pc')
    samples_2 = NestedSamples(root='./tests/example_data/pc_250')
    samples = merge_nested_samples([samples_1, samples_2])

    import matplotlib.pyplot as plt
    plt.plot((1/samples_1.nlive).cumsum(),samples_1.nlive, label='samples_1')
    plt.plot((1/samples_2.nlive).cumsum(),samples_2.nlive, label='samples_2')
    plt.plot((1/samples.nlive).cumsum(),samples.nlive, label='samples')
    plt.xlabel('compression')
    plt.ylabel('nlive')
    plt.legend()
```

![Figure_1](https://user-images.githubusercontent.com/3708486/69554026-0c5c2600-0f99-11ea-80f1-a82e200be2ee.png)